File: udp_receiver.py
import socket

# Create UDP socket
sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

# Bind to all interfaces on port 5000
sock.bind(('0.0.0.0', 5000))

print("Listening for UDP messages on port 5000...")

# Receive message
data, addr = sock.recvfrom(1024)  # buffer size 1024 bytes
print(f"Received: {data.decode()} from {addr}")

File: udp_receiver_nim/udp_receiver.nim
import net

var sock = newSocket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)
sock.bindAddr(Port(5000))

echo "Listening for UDP messages on port 5000..."

var data: string
var address: string
var port: Port
sock.recvFrom(data, 1024, address, port)
echo "Received: ", data, " from ", address, ":", port
